source/common/network/apple_dns_impl.cc

```diff
@@ -265,7 +265,9 @@ AppleDnsResolverImpl::PendingResolution::~PendingResolution() {
   }
 }
 
-void AppleDnsResolverImpl::PendingResolution::cancel() {
+void AppleDnsResolverImpl::PendingResolution::cancel(Network::ActiveDnsQuery::CancelReason) {
+  // TODO(mattklein123): If cancel reason is timeout, do something more aggressive about destroying
```

Comment:
I'd be inclined to remove the argument if it's not used anywhere, unless you actually plan on following up in the not too distant future.

Reply:
My intention here was to follow up with at least the Apple DNS provider (and maybe c-ares) to blow away the channels if we have a timeout, to make it more likely we can work around dead connections. cc @junr03 wdyt?

Reply:
@mattklein123  For the Apple implementation post https://github.com/envoyproxy/envoy/pull/17226 there is nothing to "blow away" as each PendingResolution is getting its own socket (and registered file event with the event loop). 

On the c-ares implementation, where the connection _is_ shared, the distinction would be useful. A timeout based cancelation could be treated the same as a c-ares error and used to blow up the existing channel.

Reply:
OK, let's keep it, and we can make the c-ares change for Android, which I think would be useful IMO.